File: flutter/lib/domain/ship_navigation/distances.dart
import 'dart:math';

import 'package:ship_conquest/domain/space/map_position.dart';

double calculateEuclideanDistance(MapPosition start, MapPosition end) {
  int dx = start.x - end.x;
  int dy = start.y - end.y;
  return sqrt((dx * dx + dy * dy).abs().toDouble());
}

double calculateManhattanDistance(MapPosition start, MapPosition end) {
  return (start.x - end.x).abs() + (start.y - end.y).abs().toDouble();
}

File: jvm/ship-conquest/src/main/kotlin/com/example/shipconquest/domain/ship_navigation/distances.kt
package com.example.shipconquest.domain.ship_navigation

import com.example.shipconquest.domain.Position
import kotlin.math.abs
import kotlin.math.sqrt

fun calculateEuclideanDistance(start: Position, end: Position): Double {
    val dx = start.x - end.x
    val dy = start.y - end.y
    return sqrt(abs(dx * dx + dy * dy).toDouble())
}

fun calculateManhattanDistance(start: Position, end: Position): Double {
    return abs(start.x - end.x) + abs(start.y - end.y).toDouble()
}